src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -2081,31 +2107,31 @@ LinkAnimationHeader* func_808335F4(Player* this) {
     }
 }
 
-void func_80833638(Player* this, PlayerFunc82C arg1) {
-    this->func_82C = arg1;
+void Player_SetItemActionFunc(Player* this, ItemActionFunc itemActionFunc) {
+    this->itemActionFunc = itemActionFunc;
     this->unk_836 = 0;
-    this->unk_830 = 0.0f;
+    this->skelAnimeUpperBlendWeight = 0.0f;
     func_808326F0(this);
 }
 
-void func_80833664(PlayState* play, Player* this, s8 itemAction) {
+void Player_InitItemActionWithAnim(PlayState* play, Player* this, s8 itemAction) {
     LinkAnimationHeader* current = this->skelAnime.animation;
     LinkAnimationHeader** iter = D_80853914 + this->modelAnimType;
-    u32 i;
+    u32 animGroup;
 
     this->stateFlags1 &= ~(PLAYER_STATE1_3 | PLAYER_STATE1_24);
 
-    for (i = 0; i < PLAYER_ANIMGROUP_MAX; i++) {
+    for (animGroup = PLAYER_ANIMGROUP_wait; animGroup < PLAYER_ANIMGROUP_MAX; animGroup++) {
```

Comment:
I think it is better to keep this as 0 instead of the first element of the enum. With 0 it is clear this loop is iterating from the first entry to the last one

Reply:
I suppose. I prefer the enum, but as long as we keep the name as `animGroup` instead of `i` im happy. The value gets used later outside the loop, and naming it `i` makes that less obvious

Reply:
`animGroup` is fine, yeah